service/src/main/java/com/epam/esm/service/service/user/UserService.java

```diff
@@ -1,13 +1,26 @@
 package com.epam.esm.service.service.user;
 
 import com.epam.esm.model.entity.User;
+import com.epam.esm.service.exceptions.NotFoundException;
 import com.epam.esm.service.service.EntityService;
 
+import java.util.Optional;
+
 /**
  * {@link User} service interface.
  *
  * @author Yury Zmushko
  * @version 1.0
  */
 public interface UserService extends EntityService<User> {
+
+    /**
+     * Find {@link User} by his unique login.
+     *
+     * @param login Login value.
+     * @return {@link User} with defined login.
+     */
+    User getByUniqueLogin(String login) throws NotFoundException;
```

Comment:
I planned to emphasize that this search method return not collection but one User object because of searching by unique field - login. 
I have to throw exception when nothing is found so I need pair of methods.

Reply:
Also I can have only one additional method - checkOptional, which will only check if optional is present and throw exception